src/com/facebook/buck/android/aapt/RDotTxtEntry.java

```diff
@@ -216,10 +216,24 @@ public int compareTo(RDotTxtEntry that) {
       return 0;
     }
 
-    return ComparisonChain.start()
-        .compare(this.type, that.type)
-        .compare(this.name, that.name)
-        .result();
+    ComparisonChain comparisonChain = ComparisonChain.start().compare(this.type, that.type);
+
+    String [] thisNameParts = this.name.split("_");
+    String [] thatNameParts = that.name.split("_");
+
+    int index = 0;
+    while (index < thisNameParts.length && index < thatNameParts.length) {
+      comparisonChain = comparisonChain.compare(thisNameParts[index], thatNameParts[index]);
+      index++;
+    }
+
+    if (index < thisNameParts.length) {
+      comparisonChain = comparisonChain.compare(thisNameParts[index], "");
+    } else if (index < thatNameParts.length) {
+      comparisonChain = comparisonChain.compare("", thatNameParts[index]);
+    }
```

Comment:
It looks like this is not guaranteed to consume the entire string.  What if the longer one has a double-underscore in it?  Couldn't that give a false "equal" comparison with a different string?  What about comparing the two array lengths instead of this block.  Would that work?